packages/dds/tree/src/test/shared-tree/fuzzEditGenerator.ts

```diff
@@ -0,0 +1,254 @@
+            parent: path,
+            parentField: nodeField,
+            parentIndex: nodeIndex,
+        } = getRandomPlace(tree, state.random);
+        const insert: FuzzInsert = {
+            fuzzType: "insert",
+            parent: path,
+            field: nodeField,
+            index: nodeIndex,
+            value: state.random.integer(Number.MIN_SAFE_INTEGER, Number.MIN_SAFE_INTEGER),
+        };
+        return insert;
+    }
+
+    async function deleteGenerator(state: EditState): Promise<FuzzDelete> {
+        const trees = state.testTreeProvider.trees;
+        const tree = trees[state.treeIndex];
+        // generate edit for that specific tree
+        const path = getExistingRandomNodePosition(tree, state.random);
+        return {
+            fuzzType: "delete",
+            path,
+        };
+    }
+
+    async function setPayloadGenerator(state: EditState): Promise<FuzzSetPayload> {
+        const trees = state.testTreeProvider.trees;
+        const tree = trees[state.treeIndex];
+        // generate edit for that specific tree
+        const path = getExistingRandomNodePosition(tree, state.random);
+        return {
+            fuzzType: "setPayload",
+            path,
+            value: state.random.integer(Number.MIN_SAFE_INTEGER, Number.MAX_SAFE_INTEGER),
+        };
+    }
+
+    const baseEditGenerator = createWeightedAsyncGenerator<FuzzChange, EditState>([
+        [insertGenerator, 5],
+        [deleteGenerator, 1, ({ testTreeProvider, treeIndex }) => containsAtLeastOneNode(testTreeProvider.trees[treeIndex])],
+        [setPayloadGenerator, 1, ({ testTreeProvider, treeIndex }) => containsAtLeastOneNode(testTreeProvider.trees[treeIndex])],
+    ]);
+
+    return async (state: FuzzTestState): Promise<Operation | typeof done> => {
+        const trees = state.testTreeProvider.trees;
+        // does not include last tree, as we want a passive client
+        const treeIndex = state.random.integer(0, trees.length - 2);
+        const contents = await baseEditGenerator({
+            ...state,
+            treeIndex,
+        });
+        state.numberOfEdits += 1;
+        if (contents === done) {
+            return done;
+        }
+        return { type: "edit", contents, index: treeIndex };
+    };
+};
+
+export function makeOpGenerator(): AsyncGenerator<Operation, FuzzTestState> {
+    const opWeights: AsyncWeights<Operation, FuzzTestState> = [
+        [makeEditGenerator(), 3],
+        [{ type: "synchronize" }, 1],
+    ];
+    return createWeightedAsyncGenerator(opWeights);
+}
+
+const moves = {
+    field: ["enterNode", "nextField"],
+    nodes: ["stop", "firstField"],
+};
+
+function getRandomPlace(tree: ISharedTree, random: IRandom): UpPath {
+    const testerKey: FieldKey = brand("Test");
+    const cursor = tree.forest.allocateCursor();
+    moveToDetachedField(tree.forest, cursor);
+    const firstNode = cursor.firstNode();
+    if (!firstNode) {
+        // no node exists, insert a rootnode
```

Comment:
I realize that returning the path below will lead to a root node being inserted, but this function doesn't know about that.
In order to understand this comment, a reader would have to understand how the function is used by the rest of the system (which may change over time). This makes the function less self-contained. I think the code is clear enough here, so my recommendation would be to remove the comment.

Reply:
Same for the other comment below.